torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -82,6 +87,18 @@ LTCTensorImpl::LTCTensorImpl(LazyTensor&& tensor)
   // This is a temporary fix for a PyTorch core issue,
   // according to https://github.com/pytorch/xla/pull/2682.
   is_non_overlapping_and_dense_ = false;
+
+  auto rank = tensor_->shape().Get().sizes().size();
+  std::vector<int64_t> sym_ints;
+  sym_ints.reserve(rank);
+
+  for (auto i: c10::irange(rank)) {
+    auto dim_node = MakeNode<SizeNode>(this->tensor_->GetIrValue(), i);
+    auto sn = std::make_shared<torch::lazy::SymbolicIntNode>(dim_node);
+    sym_ints.push_back(sn->toSymInt().data());
```

Comment:
If we store symints directly in sizes_, for the cases where the size is statically known, we don't want to force a symbolic int in this case, we want to store the real int. I'm not familiar enough with LTC to know how to figure this out, but this doesn't look right. (I think what's going on is you're still on the older version of the design, where we symbolic-ified all the size accesses so we could avoid having to have shape propagation formulas. But if I understand correctly, we're going to need those anyway (to propagate dynamic bit) so might as well store the actual formulas in sizes_ in that case.

Reply:
`tensor_->shape().Get()` has `is_symbolic` that tells us which dimension is truly static and which one may be dynamic. It should be safe to use upper bounds for truly static dimensions. I can make the change but I believe as of now all dimensions will be static, I'd rather keep them as expressions for now :)